addons/mrp/models/mrp_production.py

```diff
@@ -2003,10 +2003,10 @@ def action_merge(self):
         for move in self.move_finished_ids:
             dests.setdefault(move.byproduct_id.id, []).extend(move.move_dest_ids.ids)
 
-        production = self.env['mrp.production'].create({
+        production = self.env['mrp.production'].with_context(default_picking_type_id=self.picking_type_id.ids[0]).create({
             'product_id': product_id.id,
             'bom_id': bom_id.id,
-            'picking_type_id': bom_id.picking_type_id or self._get_default_picking_type(),
+            'picking_type_id': self.picking_type_id.ids[0],
```

Comment:
The `_pre_action_split_merge_hook` already checks that there is only 1 `self.picking_type_id` so you can replace `self.picking_type_id.ids[0]` with `self.picking_type_id.id` (it would be better to throw an error if for some reason there is more than 1 picking_type_id provided then to incorrectly change a MO's picking_type_id later on anyways.)

Reply:
okay, got it